sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticFieldInitializerOrder.cs

```diff
@@ -52,6 +52,31 @@ public partial interface IStaticFieldsOrder
         public const int Const2 = 5;
     }
 
+    public partial struct StaticFieldsOrderStruct
+    {
+        public static string s1 = new string('x', Const); // Compliant. Const do not suffer from initialization order fiasco.
+        public static string s2 = new string('x', Const2);
+        public static string s3 = new string('x', Y); // Noncompliant
+
+        public static Action<int> A = (i) => { var x = i + StaticFieldInitializerOrder.Y; }; // Okay??? Might or might not. For no we don't report on it
```

Comment:
I think it would be nice to have coverage in test cases also with array initialization as Corniel mentioned.